src/GeositeFramework/js/Legend.js

```diff
@@ -66,23 +68,50 @@ define(['use!Geosite',
                     return this.tmplLegendItemScale;
                 }
             }
+
             if (legend.legend.length === 1) {
                 return this.tmplLegendItemSingle;
             }
+
             return this.tmplLegendItemMultiple;
         },
 
         render: function(legendGroups) {
             var self = this,
                 $container = $('<div>');
+
             _.each(legendGroups, function(legendGroup) {
                 var service = legendGroup.service,
                     layer = legendGroup.layer,
                     legend = legendGroup.legend,
                     tmpl = self.getLayerTemplate(legend, service, layer);
-                    $container.append(tmpl(legend));
+
+                $container.append(tmpl(legend));
             });
+
             this.$el.find('.legend-body').html($container.html());
+            this.assignLegendEvents();
+        },
+
+        assignLegendEvents: function() {
+            var self = this;
+
+            $('.expand-legend').click(function(e) {
+                self.toggleExtraLegendItems(e);
+            });
+        },
+
+        toggleExtraLegendItems: function(e) {
+            var $extraLegendControl = $(e.target),
+                $extraLegendItems = $extraLegendControl.parent().parent().find('.extra');
+
+            $extraLegendItems.toggle();
+
+            if (($extraLegendItems).is(':visible')) {
+                $extraLegendControl.text('View less items');
+            } else {
+                $extraLegendControl.text('View more items');
+            }
```

Comment:
If you toggle a class on the parent instead of toggling visibility (ex. `.show-extras`), we might be able to move this text into the template.